tests/test_search.py

```diff
@@ -4,12 +4,261 @@
+    _get_api_object_names,
+    _get_capitalize_match_for_word_from_names,
+    _get_close_matches_for_word_from_names,
+    _get_exact_match_for_word_from_names,
+    _get_match_case_for_word_from_names,
+    _get_version_path_prefix_from_obj,
+    _get_wildcard_matches_for_word_from_names,
+    _search,
+    _search_semantic,
+    _search_whole_word,
+    _search_wildcard,
+)
+
+
+@pytest.mark.fluent_version("==24.2")
+def test_nltk_data_download():
+    import nltk
+
+    packages = ["wordnet", "omw-1.4"]
+    for package in packages:
+        nltk.download(package, quiet=True)
+
+    _search_semantic("读", language="cmn")
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_exact_match_for_word_from_names():
+    names = _get_api_object_names()
+    exact_match = _get_exact_match_for_word_from_names("VideoResoutionY", names)
+    assert "VideoResoutionY" in exact_match
+    assert len(exact_match) == 1
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_capitalize_match_for_word_from_names():
+    names = _get_api_object_names()
+    capitalize_match_cases = _get_capitalize_match_for_word_from_names("font", names)
+    assert "font" not in capitalize_match_cases
+    assert "Font" in capitalize_match_cases
+    assert "TextFontAutomaticHorizontalSize" in capitalize_match_cases
+    assert "TextFontAutomaticUnits" in capitalize_match_cases
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_match_case_for_word_from_names():
+    names = _get_api_object_names()
+    match_cases = _get_match_case_for_word_from_names("font", names)
+    for match_case in match_cases:
+        assert "Font" not in match_case
+        assert "font" in match_case
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_wildcard_matches_for_word_from_names():
+    names = _get_api_object_names()
+    wildcard_matches = _get_wildcard_matches_for_word_from_names("iter*", names)
+    assert "iterating" in wildcard_matches
+    assert "iterate_steady_2way_fsi" in wildcard_matches
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_get_close_matches_for_word_from_names():
+    names = _get_api_object_names()
+    close_matches = _get_close_matches_for_word_from_names("font", names)
+    assert "font" in close_matches
+
+    close_matches = _get_close_matches_for_word_from_names("fnt", names)
+    assert "font" in close_matches
+
+    close_matches = _get_close_matches_for_word_from_names("solve_flow", names)
+    assert "solve_flow_last" in close_matches
+
+    close_matches = _get_close_matches_for_word_from_names("sunshine", names)
+    assert "sunshine_factor" in close_matches
+
+
+@pytest.mark.fluent_version("==24.2")
+@pytest.mark.codegen_required
+def test_search_wildcard(capsys):
+    _search_wildcard("max*")
+    lines = capsys.readouterr().out.splitlines()
```

Comment:
This test immediately indicates a potential improvement. That is, we need to separate the evaluation of the search from the outputting of results. Have all your algorithmic functions _return_ results. Output only at the very top level.

Reply:
These functions do not return results - _search_wildcard, _search_whole_word, _search_semantic, _print_search_results

All other functions return results.

Do we need to return results in all functions ? 

We are not returning results in old search function therefore we are doing the same for new search function.

Reply:
_We are not returning results in old search function therefore we are doing the same for new search function._

Then I am proposing an improvement relative to the old code. Let's improve it in the next round, not in this PR. We can discuss separately why this point is important.

Reply:
Yes, sure.